securedrop_client/gui/widgets.py

```diff
@@ -1152,19 +1152,24 @@ def update(self):
             )
             raise
 
-    def set_snippet(self, source, uuid=None, content=None):
+    def set_snippet(self, source, uuid=None, content=None) -> None:
         """
         Update the preview snippet only if the new message is for the
         referenced source and there's a source collection. If a uuid and
         content are passed then use these, otherwise default to whatever the
         latest item in the conversation might be.
         """
-        if source == self.source.uuid and self.source.collection:
+        if source != self.source.uuid:
+            return
+
+        if self.source.collection:
             last_collection_object = self.source.collection[-1]
             if uuid and uuid == last_collection_object.uuid and content:
                 self.preview.setText(content)
             else:
                 self.preview.setText(str(last_collection_object))
+        else:
+            self.preview.setText("")
```

Comment:
preview will be hidden or already empty if there are no items in the collection so you should remove this and to reduce nesting you can check up top if the source collection is empty and return

Reply:
That was not my experience. It would not be cleared. I believe that was the reason for #922.

Reply:
So if i follow the instructions in #922: "Delete the most recent item in a conversation (I did this with a source with one file)" I do not see the snippet because it is replaced with the "Deleting" pending state which hides the preview widget that shows the snippet.

I think you might have found a different issue where we delete all items (maybe just the last item) in the source collection on the server and need to update the snippet to show empty string. Does that sound right?

Reply:
actually now i'm curious how the journalist interface handles the removal of the last item in a source's collection... i take it it doesn't remove the source? should we be able to get into a state where we have a source with no messages to show?

Reply:
> should we be able to get into a state where we have a source with no messages to show?

yep that can happen (deleting the entire source means the source can no longer login, which a journalist may want to avoid, and instead they might just want to delete old submissions individually)

Reply:
ah i see, #922 is about item deletion, which we only support in the Journalist Interface so that was implicit in the issue - this PR does fix this. i read "Delete the most recent item in a conversation (I did this with a source with one file)" as something we do in the client as in delete the most recent source, which deletes its conversation.